packages/performance/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/performance/FirebasePerformancePlugin.java

```diff
@@ -116,4 +127,236 @@ public void isEnabled(PluginCall call) {
+            result.put("value", value);
+            call.resolve(result);
+        } catch (Exception exception) {
+            Logger.error(TAG, exception.getMessage(), exception);
+            call.reject(exception.getMessage());
+        }
+    }
+
+    @PluginMethod(returnType = PluginMethod.RETURN_NONE)
+    public void record(PluginCall call) {
+        try {
+            String traceName = call.getString("traceName");
+            if (traceName == null) {
+                call.reject(FirebasePerformancePlugin.ERROR_TRACE_NAME_MISSING);
+                return;
+            }
+            Long startTime = call.getLong("startTime");
+            if (startTime == null) {
+                call.reject(FirebasePerformancePlugin.ERROR_START_TIME_MISSING);
+                return;
+            }
+            Long duration = call.getLong("duration");
+            if (duration == null) {
+                call.reject(FirebasePerformancePlugin.ERROR_DURATION_MISSING);
+                return;
+            }
+            JSObject metrics = call.getObject("metrics", new JSObject());
+            Map<String, Long> mappedMetrics = jsObjectToMetricsMap(metrics == null ? new JSObject() : metrics);
+            JSObject attributes = call.getObject("attributes", new JSObject());
+            Map<String, String> mappedAttributes = jsObjectToAttributesMap(attributes == null ? new JSObject() : attributes);
+            Trace trace = implementation.getTraceByName(traceName);
+            if (trace == null) {
+                call.reject(ERROR_TRACE_NOT_FOUND);
+                return;
+            }
+            implementation.record(trace, traceName, startTime, duration, mappedAttributes, mappedMetrics);
+            call.resolve();
+        } catch (Exception exception) {
+            Logger.error(TAG, exception.getMessage(), exception);
+            call.reject(exception.getMessage());
+        }
+    }
+
+    private static Map<String, String> jsObjectToAttributesMap(JSONObject object) throws JSONException {
+        Map<String, String> map = new HashMap<>();
+        Iterator<String> keys = object.keys();
+        while (keys.hasNext()) {
+            String key = keys.next();
+            map.put(key, object.get(key).toString());
+        }
+        return map;
+    }
+
+    private static Map<String, Long> jsObjectToMetricsMap(JSONObject object) throws JSONException {
+        Map<String, Long> map = new HashMap<>();
+        Iterator<String> keys = object.keys();
+        while (keys.hasNext()) {
+            String key = keys.next();
+            if (object.get(key) instanceof Long) {
+                map.put(key, (Long) object.get(key));
+            } else if (object.get(key) instanceof Double) {
+                map.put(key, (long) Math.floor((double) object.get(key)));
+            } else {
+                throw new JSONException(FirebasePerformancePlugin.ERROR_INVALID_METRIC_VALUE);
+            }
+        }
+        return map;
```

Comment:
Please move this methods into a new `FirebasePerformanceHelper` class ([example](https://github.com/capawesome-team/capacitor-firebase/blob/258ea28d64525777850e952f8437f6526d29c480/packages/firestore/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/firestore/FirebaseFirestoreHelper.java#L24)). I like to use this approach to keep the plugin classes as small as possible.

Reply:
Don't we need the same methods for iOS?

Reply:
On swift it was much more compact to iterate over a dictionary that why i did not create a separate method :)). But after removing `.record(...)` these methods will be removed as well.